seismicpro/src/utils/interactive_plot_utils.py

```diff
@@ -0,0 +1,367 @@
+        return titled_box
+
+    def _resize(self, width):
+        width += 4  # Correction for main axes margins
+        self.header.layout.width = f"{int(width)}px"
+
+    def on_resize(self, event):
+        """Resize the plot on the `fig` canvas size change."""
+        self._resize(event.width)
+
+    def _click(self, coords):
+        coords = self.click_fn(coords)
+        if coords is None:  # Ignore click
+            return
+        if self.click_marker is not None:
+            self.click_marker.remove()
+        self.click_marker = self.ax.scatter(*coords, **self.marker_params, zorder=10)
+        self.fig.canvas.draw_idle()
+
+    def on_click(self, event):
+        """Remember the mouse button click time to further distinguish between mouse click and hold events."""
+        # Discard clicks outside the main axes
+        if event.inaxes != self.ax:
+            return
+        if event.button == 1:
+            self.click_time = time()
+
+    def on_release(self, event):
+        """Handle the mouse button click event if it was short enough to consider it as a single click."""
+        # Discard clicks outside the main axes
+        if event.inaxes != self.ax:
+            return
+        if event.button == 1 and ((time() - self.click_time) < MAX_CLICK_TIME):
+            self.click_time = None
+            self._click((event.xdata, event.ydata))
+
+    def _unclick(self):
+        if self.click_marker is None:
+            return
+        self.unclick_fn()
+        self.click_marker.remove()
+        self.click_marker = None
+        self.fig.canvas.draw_idle()
+
+    def on_press(self, event):
+        """Undo mouse button click on ESC key press if allowed."""
+        if (event.inaxes == self.ax) and (event.key == "escape") and self.is_unclickable:
+            self._unclick()
+
+    def set_view(self, view):
+        """Set the current view of the plot to the given `view`."""
+        if view < 0 or view >= self.n_views:
+            raise ValueError("Unknown view")
+        self.current_view = view
+        self.redraw()
+
+    def on_view_toggle(self, event):
+        """Switch the plot to the next view."""
+        _ = event
+        if self.is_unclickable:
+            self._unclick()
+        self.set_view((self.current_view + 1) % self.n_views)
+
+    def set_title(self, title=None):
+        """Update the plot title. If `title` is not given, the default title of the current view is used."""
+        title = title or self.title
+        self.title_widget.value = TITLE_TEMPLATE.format(style=TITLE_STYLE, title=title)
+
+    def clear(self):
+        """Clear the plot axes and revert them to the initial state."""
+        # Remove all axes except for the main one if they were created (e.g. a colorbar)
+        for ax in self.fig.axes:
+            if ax != self.ax:
+                ax.remove()
+        self.ax.clear()
+        # Stretch the axes to its original size
```

Comment:
reset axis aspect to the mode that is appropriate for most plotting methods

```suggestion
        self.ax.clear()
        self.ax.set_aspect('auto')
        # Stretch the axes to its original size
```

Reply:
Fixed